app/services/concerns/wiki_client.rb

```diff
@@ -4,9 +4,9 @@
 module WikiClient
   extend ActiveSupport::Concern
 
-  WIKI_SITE = ENV.fetch('WIKIDATA_SITE')
-  WIKI_USERNAME = ENV.fetch('WIKIDATA_USERNAME')
-  WIKI_PASSWORD = ENV.fetch('WIKIDATA_PASSWORD')
+  WIKI_SITE = ENV['WIKIDATA_SITE']
+  WIKI_USERNAME = ENV['WIKIDATA_USERNAME']
+  WIKI_PASSWORD = ENV['WIKIDATA_PASSWORD']
```

Comment:
I might be missing something obvious, but if the problem is that these are evaluated when the module is loaded, couldn't you add private module methods like:

```ruby
  def wiki_site
    ENV.fetch('WIKIDATA_SITE')
  end
```

I know in Slack that you said your refactoring of this got a bit involved, but the above seems like a small change that still preserves the behaviour where you get a really obvious error if those environment variables aren't present.

(BTW, small typo in the commit message - I think you meant "when running the tests on Travis")

Reply:
Ah yes, preserving the existing behaviour is probably preferable. Have re-done that commit as ff221767e7e117b9f21460592f500a44cbf81a29 and updated the commit message to reference Travis rather than Heroku.